### PR TITLE
Blog Post - How to cite R and R packages

### PR DESCRIPTION
**Pull request by steffilazerte**
* [X] I have read the Content Guidelines.
* [X] I have read the Technical Guidelines.
* [X] I used or followed the R Markdown or Markdown template.
* [X] I have followed the Style Guide.
* [X] I created or updated my author metadata with correct folder name.
* [X] I have added relevant tags after browsing existing tags (incuding "community" tag).
* [X] I have added the "tech notes" tag if this is a technote.
* [X] I ran `roblog::ro_lint_md()` on index.md (optional).
* [X] I ran `roblog::ro_check_urls()` on index.md (optional).
* [X] I ran a spell-check on index.md.

### CODE REVIEW

**Comment by steffilazerte** [comment]
@maelle did you want to add anything (and then add yourself as co-author), or review? I think @stefaniebutland said she'd do a quick review too (?).

I feel like parts are a bit abrupt, but I think I could use some external input. 

If it's looking good, I could probably post for Tuesday (Nov 9th), Stef, so we keep it to one post per week?

**Comment by stefaniebutland** [comment]
@steffilazerte please DM me on Slack to do a quick review when ya'll are both happy with it

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+This is a tricky question, and to be honest, I'm not really sure of the best answer.
+
+I always advise citing statistical packages, no question, and any package that is specific to a scientific domain or methodology (i.e. if you used [magick](https://docs.ropensci.org/magick) to process images before analysis,
+or if you used [tidyhydat](https://docs.ropensci.org/tidyhydat) to retrieve hydrology data, definitely cite those packages). 
+
+The tricky bit is packages that are used generally, like data munging packages, 
+or packages like [osfr](https://docs.ropensci.org/osfr) which are used as part of the scientific process 
+(i.e. connecting to OSF, the Open Science Framework), 
+but not necessarily for the analysis specifically.
+In an ideal world, everything we use would be cited, but with word limits and editors less aware of the importance of citing software, it's often hard to justify citing everything in a manuscript.
```
Comment (maelle) [inline content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:64]:
```suggestion
In an ideal world, everything we use would be cited, but with word and references limits and editors less aware of the importance of citing software, it's often hard to justify citing everything in a manuscript. 
```

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+I always advise citing statistical packages, no question, and any package that is specific to a scientific domain or methodology (i.e. if you used [magick](https://docs.ropensci.org/magick) to process images before analysis,
+or if you used [tidyhydat](https://docs.ropensci.org/tidyhydat) to retrieve hydrology data, definitely cite those packages). 
+
+The tricky bit is packages that are used generally, like data munging packages, 
+or packages like [osfr](https://docs.ropensci.org/osfr) which are used as part of the scientific process 
+(i.e. connecting to OSF, the Open Science Framework), 
+but not necessarily for the analysis specifically.
+In an ideal world, everything we use would be cited, but with word limits and editors less aware of the importance of citing software, it's often hard to justify citing everything in a manuscript. 
+
+My advice is to cite every package you think you can get away with, especially if it was a major part of your process.
```
Comment (maelle) [inline content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:66]:
```suggestion
My advice is to cite every package you think you cannot get away with, especially if it was a major part of your process.
```

?

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+Anything else, try to make sure it's prominently displayed in your scripts and if possible include your scripts as supplemental to the manuscript. This way any curious readers will be exposed to the packages if nothing else. For packages like osfr, you could share your OSF page/doi, and perhaps mention that it was managed with osfr.
```
Comment (maelle) [inline content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:67]: Maybe re-underline that having the packages mentioned in supplementary materials isn't enough?

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+> All analysis were performed using R Statistical Software 
+> (v`r version$major`.`r version$minor`; R Core Team 2021).
+> Temperature data was obtained from Environment and Climate Change Canada
+> via the weathercan R package (v`r packageVersion("weathercan")`; LaZerte and Albers 2018).
+> Vegetation photos were simplified and processed prior to analysis using the magick R package (v`r packageVersion("magick")`; Ooms 2021).
+
+
+## Reproducible workflow for managing citations
+
+Being ready to cite the packages you've used will be so much easier if you keep
```
Comment (maelle) [inline content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:128]: Maybe mention that using a workflow tool like targets can help keep track of packages used as everything will be neatly organized?

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+Consider keeping compiled reports of your analysis scripts 
+(either rendering your scripts with the [RStudio's Compile Report](https://rmarkdown.rstudio.com/articles_report_from_r_script.html) button,
+'spinning' your scripts via rmarkdown and knitr[^1], 
+or using Rmd files and knitting them.)
+
+This way you can include a call to `sessionInfo()` or `devtools::session_info()`,
+at the end of your script. 
+
+```{r}
+devtools::session_info()
```
Comment (maelle) [inline content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:140]: but note this includes dependencies of dependencies, that one will generally _not_ cite?

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+```
+
```
Comment (maelle) [inline content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:142]: Mention renv?

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+c("weathercan", "magick", "tidyhydat") %>%
+  map(citation) %>%
+  print(style = "text")
+```
+
+
+## Doing more with citations
+
+- If you want to go next level and start managing your citations in R, checkout
```
Comment (maelle) [inline content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:167]: Also note the tweet https://twitter.com/__jsta/status/1258419526636625920 that mentioned stuff missing from the post

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+library(purrr)
+
+c("weathercan", "magick", "tidyhydat") %>%
+  map(citation) %>%
+  print(style = "text")
+```
+
+
+## Doing more with citations
+
```
Comment (maelle) [inline content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:166]: Also

* https://github.com/Pakillo/grateful
* https://indrajeetpatil.github.io/awesome-r-pkgtools/#gratitude-

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+- Are you a package developer who would like to see their packages cited more readily?
+Check out the post [Make Your R Package Easier to Cite](blog/2021/02/16/package-citation/) 
+for tips and tricks.
+
+## Final thoughts
+
+Free and Open Source Software (FOSS) is a magical thing. 
+It equalizes financial disparity among scientific institutions and allows the
+development of highly specialized analysis in really specific domains. 
+Citing R and R packages is not only important for reproducible science, but is
```
Comment (maelle) [inline content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:181]: I don't think citing packages will be crucial for reproducibility but the point could be made that you want to track packages both for citing them and for reproducibility?

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+oh so meaningful to software developers. 
+
+If we want them to keep doing what they do, the least we can do is cheer them on!
+
+
```
Comment (maelle) [inline content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:186]: Maybe add a question e.g. inviting readers to share their tips or challenges? Not sure it really triggers more engagement though.

**Review by maelle — CHANGES_REQUESTED** [review]
Thanks a lot @steffilazerte for writing this important & engaging post! I made some comments, the main one being that I wouldn't put reproducibility first as citing packages isn't directly optimal for this.

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+## Doing more with citations
+
+- If you want to go next level and start managing your citations in R, checkout
+the post [A Roundup of R Tools for Handling BibTeX](/blog/2020/05/07/rmd-citations/) 
+
+- Stay tuned for an upcoming blog post on the newly added [cffr](https://docs.ropensci.org/cffr) package for working with the CFF citation format for your package or in general.
+
+- Are you a package developer who would like to see their packages cited more readily?
+Check out the post [Make Your R Package Easier to Cite](blog/2021/02/16/package-citation/)
```
Comment (maelle) [inline content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:173]:
```suggestion
Check out the post [Make Your R Package Easier to Cite](/blog/2021/02/16/package-citation/) 
```

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd [hunk 12]
```diff
@@ -0,0 +1,195 @@
...
+
+Many scientists don't know that they should be citing R packages let alone R, 
+and, if they do know, they often struggle with how.
+So here's a short primer on why and how to get started!
+
+## Why cite?
+
+It's extremely important to cite both R and R packages for several reasons:
+
+1) **Reproducible Science**
```
[the remark this answers is not in the document]
Comment (maelle) [reply content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:32]: Reg Docker a reference might be https://journal.r-project.org/archive/2020/RJ-2020-007/index.html

[the remark this answers is not in the document]
Comment (steffilazerte) [reply content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:32]: True, I think I've been thinking about the audience of this post being scientists who might just be able to share their scripts and compile them into html reports, but for whom renv would be a bit much. 

It's definitely a good point, I think I will shuffle things around and will point out that if you want to get serious, you can explore renv or docker :grin:

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd (hunk 2 above)
Comment (steffilazerte) [reply content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:66]: Not quite the meaning I wanted. I was referring to getting away with the editor not complaining :grin:

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd (hunk 3 above)
Comment (steffilazerte) [reply content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:67]: Good point, this is making the best of a bad situation, not something to aim for!

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd (hunk 8 above)
Comment (steffilazerte) [reply content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:166]: SO COOL!!!!

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd (hunk 9 above)
Comment (steffilazerte) [reply content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:181]: Perhaps not strictly for *reproducibility*, but for *repeatability*?

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd (hunk 12 above)
[the remark this answers is not in the document]
Comment (steffilazerte) [reply content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:32]: Also, I realize that I've confused repeatable with reproducible. Fixing that!

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd [hunk 13]
```diff
@@ -0,0 +1,195 @@
...
+It's extremely important to cite both R and R packages for several reasons:
+
+1) **Reproducible Science**  
+If it isn't clear how you did a thing or how/where you got your data, your work isn't reproducible. 
+Software (like R), R packages, and their versions are important information to include in a manuscript.
+It not only helps other scientists understand and reproduce your work, but helps share important and useful tools. 
+Including versions is also really important as a way of recording the context of your manuscript 
+when software changes.
+
+2) **Developer Credit**
```
[the remark this answers is not in the document]
Comment (steffilazerte) [reply content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:39]: So funny yet sad! A great footnote to add

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd (hunk 6 above)
Comment (steffilazerte) [reply content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:142]: Added renv to the end along with references to docker

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd (hunk 7 above)
Comment (steffilazerte) [reply content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:167]: Good idea

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd (hunk 9 above)
Comment (steffilazerte) [reply content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:181]: I've clarified this throughout

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd (hunk 10 above)
Comment (steffilazerte) [reply content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:186]: I think I'll leave it as is. Might make a good tweet comment though?

**Comment by steffilazerte** [comment]
Thanks for the review @maelle! Really great point about reproducibility, made me realize that I mixed up reproducible and repeatable 🤦🏼 . Thanks for all the extra info in links etc. it really fleshes the post out!

@stefaniebutland I think it's ready for your quick review, after that @maelle can do her final check Tuesday morning (if you can!) and then I'll publish *my* Tuesday morning :grin: Sorry to make everything so last minute!

-----
Tweet draft:

[blog] How to Cite R and R Packages 📦

🙏🏻 Developers need our love, gratitude, and most important, our citations! 

🤔 Why does this matter? How do you start?

@steffilazerte's primer includes how to find citation info, wordings, and workflows. 

https://ropensci.org/blog/2021/11/16/how-to-cite-r-and-r-packages/

#RStats

**Review by stefaniebutland — COMMENTED** [review]
It's not last-minute Steffi - you were planning to do this for the 23rd :-)

I really like the motivation in the intro.

1.
>  A Roundup of R Tools for Handling BibTeX[^4]

fix citation formatting 


2. I feel like it's important to list the more "academic" reason for citation (repeatable science) before listing gratitude cause it's not primarily about being nice.

3. I think it's very important to look at, maybe paraphrase, and definitely cite existing work on software citation. See [FORCE11 Software Citation Implementation Working Group](https://www.force11.org/group/software-citation-implementation-working-group) and links therein, including [Software citation principles](https://peerj.com/articles/cs-86/) (2016) and [Software Citation Checklist for Authors](https://zenodo.org/record/3479199#.YYmfT73MKAk)(2019) which has some specifics under "What software should I cite?". Doing that puts this post in bigger context and  professionalizes your recommendation to cite software. I just noticed that the author of CFF format, Stephan Druskat, is an author on that document.

**Comment by steffilazerte** [comment]
Thanks @stefaniebutland, better? Or should I give it more of an in-depth consideration?

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd [hunk 14]
```diff
@@ -0,0 +1,222 @@
...
+Many developers of R packages for science are themselves scientists.
+They may be developing packages in their own time, or are trying to fit it into a busy schedule. 
+Without citations, most scientists don't get credit for their work. 
+This makes it hard to prioritize work on open software projects. 
+If we want to ensure that high-quality open software continues, 
+it's important that the developers get credit for their work[^1].
+
+2) **Repeatable Science**  
+If it isn't clear how you did a thing or how/where you got your data, your work isn't repeatable. 
+The lack of citations of software in scientific publications has become a problem to the extent that working groups, such as [FORCE11 Software Citation Working Group](https://www.force11.org/group/software-citation-implementation-working-group), have been created to establish guidelines[^2] and standards[^3] for citing software.
```
Comment (stefaniebutland) [inline content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:43]:
```suggestion
The lack of citations of software in scientific publications has become a problem to the extent that working groups, such as [FORCE11 Software Citation Implementation Working Group](https://www.force11.org/group/software-citation-implementation-working-group), have been created to establish guidelines[^2] and standards[^3] for citing software.
```

**Review by stefaniebutland — APPROVED** [review]
The addition LGTM and perhaps @maelle will have deeper thoughts on it :-)

I added one word to name of working group

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd (hunk 14 above)
Comment (steffilazerte) [reply content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd]: Done!

**Comment by steffilazerte** [comment]
Great thanks! @maelle, on to you, let me know what you think when you're back at your desk (if you have time!)

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd [hunk 15]
```diff
@@ -0,0 +1,222 @@
...
+This makes it hard to prioritize work on open software projects. 
+If we want to ensure that high-quality open software continues, 
+it's important that the developers get credit for their work[^1].
+
+2) **Repeatable Science**  
+If it isn't clear how you did a thing or how/where you got your data, your work isn't repeatable. 
+The lack of citations of software in scientific publications has become a problem to the extent that working groups, such as [FORCE11 Software Citation Implementation Working Group](https://www.force11.org/group/software-citation-implementation-working-group), have been created to establish guidelines[^2] and standards[^3] for citing software.
+Software (like R), R packages, and their versions are important information to include in a manuscript.
+It not only helps other scientists understand and repeat your work, but helps share important and useful tools. 
+Including versions is also really important as a way of recording the context of your manuscript
```
Comment (maelle) [inline content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:46]: I'd still put this last and underline that citations is not a good way to record such information.

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd [hunk 16]
```diff
@@ -0,0 +1,222 @@
...
+or packages like [osfr](https://docs.ropensci.org/osfr) which are used as part of the scientific process 
+(i.e. connecting to [OSF, the Open Science Framework](https://osf.io/)), 
+but not necessarily for the analysis specifically.
+In an ideal world, everything we use would be cited, but with word and reference limits and editors less aware of the importance of citing software, it's often hard to justify citing everything in a manuscript. 
+
+The general advice by the by the [FORCE11 Software Citation Implementation Working Group](https://peerj.com/articles/cs-86/#discussion) is to include software important to the research outcome. 
+I would also add that it's not a bad thing to cite open-source software that was a major part of your workflow (for the purposes of credit, if not repeatability). 
+Anything else, try to make sure it's prominently displayed in your scripts and if possible include your scripts as supplemental to the manuscript. This way any curious readers will be exposed to the packages if nothing else. For packages like osfr, you could share your OSF page/doi, and perhaps mention that it was managed with osfr.
```
Comment (maelle) [inline content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:74]:
```suggestion
Anything else, try to make sure it's prominently displayed in your scripts and if possible include your scripts as supplemental to the manuscript. This way any curious readers will be exposed to the packages if nothing else. For packages like osfr, you could share your OSF page/DOI, and perhaps mention that it was managed with osfr.
```

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd (hunk 4 above)
Comment (maelle) [reply content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:136]: Re-upping this comment :grin:

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd [hunk 17]
```diff
@@ -0,0 +1,222 @@
...
+```
+
+
+You may also consider using the `cite_packages()` function from the super cool [grateful](https://github.com/Pakillo/grateful) package to create a formatted bibliography of all the packages used in a project.
+
+
+Finally, if you're getting **serious** about ensuring your work is not only *repeatable* but
+also *reproducible*[^5], you might want to check out R packages that help control
+package versions like [renv](https://rstudio.github.io/renv/), 
+or [Docker images](https://journal.r-project.org/archive/2020/RJ-2020-007/index.html),
```
Comment (maelle) [inline content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:179]:
```suggestion
or [Docker containers](https://journal.r-project.org/archive/2020/RJ-2020-007/index.html),
```

That said I am not entirely sure it's the right change

**Review by maelle — COMMENTED** [review]
A few last comments, great post! :sparkles:

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd (hunk 15 above)
Comment (steffilazerte) [reply content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:46]: I don't think I agree, now that we're talking about *repeatable* not *reproducible* science. Repeatable means you have enough information in the manuscript to repeat the general work, not that you can exactly reproduce the analysis. 

The FORCE11 Software Citation Implementation Working Group [gets at this a bit](https://peerj.com/articles/cs-86/#p-55) when they discuss provenance (reproducibility) vs. citation (repeatability).

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd (hunk 17 above)
Comment (steffilazerte) [reply content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:179]: I think you're right (https://phoenixnap.com/kb/docker-image-vs-container)

File: content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd (hunk 15 above)
Comment (maelle) [reply content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:46]: Ok, I can see your point!

Comment (steffilazerte) [reply content/blog/2021-11-16-how-to-cite-r-and-r-packages/index.Rmd:46]: I've added a note here, clarifying repeatable vs. reproducible :grin:

**Pull request merged**